internal/common/db.go

```diff
@@ -199,6 +201,16 @@ func GetHostFromDBWhere(db *gorm.DB, where ...interface{}) (*Host, error) {
 	return &host, nil
 }
 
+func GetInfraEnvFromDB(db *gorm.DB, infraEnvID strfmt.UUID) (*InfraEnv, error) {
+	var infraEnv InfraEnv
+
+	err := db.First(&infraEnv, "id = ?", infraEnvID.String()).Error
+	if err != nil {
+		return nil, errors.Wrapf(err, "failed to get infra env %s", infraEnvID)
```

Comment:
Wrapping the error will hide the original error, i suggest to make check if the error is not found and return `ApiError` or the original error.

Reply:
this is an infrastructure function. I think it is more appropriate to make the decision (ApiError| original error) at the upper level. this is the way GetClusterFromDB is also implemented, and the examination can be done by using errors.Is

Reply:
Does error is work on a wrapped error?

Reply:
yes

Reply:
// Is reports whether any error in err's chain matches target.
//
// The chain consists of err itself followed by the sequence of errors obtained by
// repeatedly calling Unwrap.

Reply:
The InfraEnv AI service resource to not be created until the discovery ISO is first downloaded for a given cluster. As a result, the ignition download API endpoints are not available following a single call to the cluster creation API. This is problem for my Ansible create-cluster tasks, I am downloading the ignition before the discovery iso. What steps do I need to change in my tasks with this PR?

Reply:
Thanks for the comment, @yevgeny-shnaidman will handle this issue in a different PR.

Reply:
Thank you, I want to make clear also that I need to patch the discovery iso ignition before downloading it, so downloading it twice woudln't make much sense